synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+
+        Arguments:
+            synapse_agent_session: The response from the REST API.
+
+        Returns:
+            The AgentSession object.
+        """
+        self.id = synapse_agent_session.get("sessionId", None)
+        self.access_level = synapse_agent_session.get("agentAccessLevel", None)
+        self.started_on = synapse_agent_session.get("startedOn", None)
+        self.started_by = synapse_agent_session.get("startedBy", None)
+        self.modified_on = synapse_agent_session.get("modifiedOn", None)
+        self.agent_registration_id = synapse_agent_session.get(
+            "agentRegistrationId", None
+        )
+        self.etag = synapse_agent_session.get("etag", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Start_Session: {self.id}"
+    )
+    async def start_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        """Starts an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The new AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session_response = await start_session(
+            access_level=self.access_level.value,
+            agent_registration_id=self.agent_registration_id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(session_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Session: {self.id}"
+    )
+    async def get_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        """Gets an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The retrieved AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session_response = await get_session(
+            id=self.id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(synapse_agent_session=session_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Update_Session: {self.id}"
+    )
+    async def update_async(
+        self,
+        *,
+        access_level: AgentSessionAccessLevel,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "AgentSession":
+        """Updates an agent session. Only updates to the access level are currently supported.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The updated AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+
+        self.access_level = access_level
```

Comment:
My expectation was that folks would be updating attributes on the instance and then calling the store/update functions, rather than passing the updated values into those functions.

My idea is that I wanted any arguments you pass into the function to determine how it would execute, rather than the data it would be acting on.

In short the idea being:
```
agent_session.access_level = new_value
agent_session.update()
```

Reply:
I think using `update` here is fine because it's the terminology used in the [API docs](https://rest-docs.synapse.org/rest/PUT/agent/session/sessionId.html). Plus an Agent session isn't a Synapse Entity so conceptually it's different to interact with.